### ISSUE TITLE
Improve pylint compatibility

### ISSUE THREAD

**Issue by tcmitchell** [SynBioDex/pySBOL3#433]
Get on the road to supporting pylint in the GitHub Actions. Current score is -36.20/10. We can do better.

Of note, pylint does not like all the wildcard imports. There are more than 9,000 wildcard import issues.

**Comment by GeneCodeSavvy** [comment]
Hello everyone,

Can I work on this issue? I realize that addressing over 9,000 wildcard import issues—and potentially many other linting problems—is a significant undertaking. However, I see this as a fantastic opportunity to deeply familiarize myself with the codebase.

I’m committed to seeing this through. I am already familiar with Python, Git and Github. Understanding this codebase and eventually pySBOL3 would greatly help me when I start with utility function for converting between SBOL2 and SBOL3.

Best regards,  
Harsh

**Comment by tcmitchell** [comment]
Hi @GeneCodeSavvy, you are welcome to work on this issue. My **_strong_** recommendation is that you make iterative progress. Try to tackle one pylint issue at a time. This will reduce the set of disabled checks. Start with the easy ones and work towards the harder ones. Submit PRs on a regular basis. Don't make too many changes in a PR, they will be too hard to review. Start by fixing just one disabled pylint warning per PR. I would prefer 10 PRs, each with one pylint warning fix, to 1 PR with intermingled changes fixing 10 pylint warnings. The latter will be very hard to review.

Make use of this command, slightly modified from what you see in `setup.cfg`:

`pylint --rcfile=/dev/null --reports=y --max-line-length=120 sbol3 test`

Note: this assumes Linux or macOS. I don't know what the Windows equivalent of `/dev/null` is. 

This will generate a report and show what pylint warnings have just a few occurrences, and what pylint warnings have lots of occurrences.

**Comment by tcmitchell** [comment]
@GeneCodeSavvy  you asked on https://github.com/SynBioDex/pySBOL3/pull/465 for next steps. There are still a lot of pylint warnings. Were you asking what pylint warnings should be tackled next, or were you seeking a task other than pylint work?

**Comment by GeneCodeSavvy** [comment]
@tcmitchell Yes, I wanted to know about our priorities for pylint warnings because I’ve already addressed all the easy fixes (please let me know if I have missed any)

**Comment by tcmitchell** [comment]
Here are some that might be easy. These are guesses, you'll have to judge for yourself once you start looking at them:
* unused-variable
* unnecessary-pass
* useless-return
* unspecified-encoding
* isinstance-second-argument-not-valid-type (use caution here, it might be complicated)
* useless-parent-delegation

I appreciate all the work that you are doing to improve the code!

**Comment by GeneCodeSavvy** [comment]
Thank you, I'll work on them this weekend.

**Comment by GeneCodeSavvy** [comment]
Hello @tcmitchell, I’ve completed all the easy fixes. The last two of your recommendations would require some not-so-straightforward refactoring.
I’d like to know which one I should tackle next. 

I’ve also read the SBOL2 and SBOL3 specifications and the SBOL_NOTEBOOKS repository. If you have any other tasks I can look into, I’m open to doing so.

Regards,
Harsh Sharma

**Comment by tcmitchell** [comment]
Hi @GeneCodeSavvy , thank you for the PRs to make progress on this issue. It is greatly appreciated!

I don't have specific issues in this repository to point you at. I have browsed and many of them seem rather old which makes me believe they are not pressing issues.

Here are a few possibilities of things you could do if you're interested:

- There is an effort in the SBOL community to work on the SBOL2 to SBOL3 translator. That's going on in the [SynBioDex/SBOL-utilities](https://github.com/SynBioDex/SBOL-utilities) repository and people like @bbartley and @Gonza10V have been involved. I'm sure they would appreciate another set of hands.
- The SBOL Editors set up [SynBioDex/SBOL-Notebooks](https://github.com/SynBioDex/SBOL-Notebooks) to teach people and to help new people get started with SBOL. You may be able to help out there. Folks like @PrashantVaidyanathan and @Gonza10V are good points of contact.
- You are welcome to browse the [SynBioDex/pySBOL3](https://github.com/SynBioDex/pySBOL3) issues and pick one out. Many are a little sticky, meaning a deeper understanding may be needed to tackle them. I suspect the easier ones got attention first, and there aren't too many of them any more.

**Comment by GeneCodeSavvy** [comment]
@tcmitchell Thank you for the suggestions! I’d love to contribute to the SBOL2 to SBOL3 translator effort in the **SynBioDex/SBOL-utilities** repository. I’ll start by exploring the repo, reviewing the existing implementation, and identifying areas where I can contribute.  

I'll also reach out to @bbartley and @Gonza10V to understand the current priorities and how I can be most helpful.

Thank you for your guidance!  

Best,  
Harsh
